src/diffpy/utils/transforms.py

```diff
@@ -121,17 +122,49 @@ def tth_to_q(tth, wavelength):
     return q
 
 
-def q_to_d(qarray):
-    return 2.0 * np.pi / copy(qarray)
+def q_to_d(q):
+    r"""
+    Helper function to convert q to d on independent variable axis, using :math:`d = \frac{2 \pi}{q}`.
+
+    Parameters
+    ----------
+    q : 1D array
+        The array of :math:`q` values np.array([qs]).
+        The units of q must be reciprocal of the units of wavelength.
+
+    Returns
+    -------
+    d : 1D array
+        The array of :math:`d` values np.array([ds]).
+    """
+    if 0 in q:
+        raise ValueError(invalid_input_emsg)
```

Comment:
just make this `np.inf` but also give a warning message that the largest d is infinite and can't be plotted?  Something like that?